types/block.go

```diff
@@ -602,19 +602,21 @@ const (
 
 // CommitSig is a part of the Vote included in a Commit.
 type CommitSig struct {
-	BlockIDFlag      BlockIDFlag `json:"block_id_flag"`
-	ValidatorAddress Address     `json:"validator_address"`
-	Timestamp        time.Time   `json:"timestamp"`
-	Signature        []byte      `json:"signature"`
+	BlockIDFlag      BlockIDFlag         `json:"block_id_flag"`
+	ValidatorAddress Address             `json:"validator_address"`
+	Timestamp        time.Time           `json:"timestamp"`
+	Signature        []byte              `json:"signature"`
+	VoteExtension    VoteExtensionSigned `json:"vote_extension"`
```

Comment:
off topic question: how would this work with aggregation of signatures?

Reply:
Signature aggregation for consensus votes should be managed by Tendermint.

If BLS was being used, and your state machine wanted VoteExtensions, then you would have to pay in the cost of verifying BLS signatures across many distinct messages. If VoteExtensionSigned was left blank, then all messages would still be the same.